packages/services/f-braze-adapter/src/services/BrazeConsumer.js

```diff
@@ -0,0 +1,147 @@
+import uniq from 'lodash.uniq';
+import {
+    filterByBrands,
+    filterByEnabledCardTypes,
+    pipe,
+    filterByCurrentlyActive,
+    sortByCardOrder
+} from './utils/index';
+import InvalidConsumerConfigError from './errors/InvalidConsumerConfigError';
+import { LogService } from './logging/logging.service';
+
+class BrazeConsumer {
+    /**
+     * Setter that is called when cards are set on the class, applies any filter methods on those cards and
+     * sets the cards variable
+     * @param cards
+     */
+    set cards (cards) {
+        this._cards = this.applyFilterMethods(cards);
+    }
+
+    constructor ({
+        enabledCardTypes,
+        brands = [],
+        callbacks,
+        interceptInAppMessages = {},
+        interceptInAppMessageClickEvents = {},
+        loggerCallbacks = {},
+        customFilters = []
+    }) {
+        this.defaultEnabledCardTypes = [
+            'Home_Promotion_Card_1',
+            'Home_Promotion_Card_2',
+            'Post_Order_Card_1',
+            'Promotion_Card_1',
+            'Promotion_Card_2'
+        ];
+
+        // check list all must be true to instantiate
+        this.optionsChecks = [
+            { 'Consumer callbacks must be an Object': !Array.isArray(callbacks) },
+            { 'Consumer callbacks must contain at least ONE callback function': callbacks && Object.keys(callbacks).length !== 0 },
+            { 'Consumer interceptInAppMessages must be an Object': !Array.isArray(interceptInAppMessages) },
+            { 'Consumer interceptInAppMessageClickEvents must be an Object': !Array.isArray(interceptInAppMessageClickEvents) },
+            { 'Consumer customFilters must be an Array': Array.isArray(customFilters) },
+            { 'Consumer brands must be an Array': Array.isArray(brands) }
+        ];
+
+        this.optionsChecks.forEach(check => {
+            Object.keys(check).forEach(key => {
+                if (!check[key]) {
+                    throw new InvalidConsumerConfigError(key);
+                }
+            });
+        });
+
+        this.enabledCardTypes = enabledCardTypes || this.defaultEnabledCardTypes;
```

Comment:
Can these be written more succinctly with rest operators?

Reply:
I not sure, if your thinking of spread operator then no as it needs to be one or the other.

Reply:
I was thinking of that.